File: moban_java/src/main/java/org/jdk/project/dto/ListResponse.java
package org.jdk.project.dto;

import java.util.List;
import lombok.Getter;

/** 通用列表返回。 */
@Getter
public class ListResponse<T> {
  private final long total;
  private final List<T> rows;

  public ListResponse(long total, List<T> rows) {
    this.total = total;
    this.rows = rows;
  }

  public static <T> ListResponse<T> of(long total, List<T> rows) {
    return new ListResponse<>(total, rows);
  }
}

File: backend/src/main/kotlin/org/jdk/project/dto/ListResponse.kt
package org.jdk.project.dto

/** 通用列表返回。 */
class ListResponse<T>(
  val total: Long,
  val rows: List<T>,
) {
  companion object {
    @JvmStatic
    fun <T> of(total: Long, rows: List<T>): ListResponse<T> {
      return ListResponse(total, rows)
    }
  }
}
